Compilation Issue - Implicit-fallthrough

Compiler version used:
clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

I had downloaded the Swiftshader sources and followed the notes listed in the below link:
https://swiftshader.googlesource.com/SwiftShader/+/refs/heads/master
Compilation failed SPIRV-Tools for the below : 

SwiftShader/third_party/SPIRV-Tools/source/val/validate_cfg.cpp: In function ‘spv_result_t spvtools::val::CfgPass(spvtools::val::ValidationState_t&, const spvtools::val::Instruction*)’:                                                                                                                                                              
SwiftShader/third_party/SPIRV-Tools/source/val/validate_cfg.cpp:1092:33: error: this statement may fall through [-Werror=implicit-fallthrough=]            
                << "return type.";                                                                                                                                                          
                                 ^                                                                                                                                                          
/SwiftShader/third_party/SPIRV-Tools/source/val/validate_cfg.cpp:1095:5: note: here                                                                          
     case SpvOpKill:                                                                                                                                                                        
     ^~~~                                                                                                                                                                                  
cc1plus: all warnings being treated as errors                                                                                                                                              
third_party/SPIRV-Tools/source/CMakeFiles/SPIRV-Tools-shared.dir/build.make:540: recipe for target 'third_party/SPIRV-Tools/source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_cfg.cpp.o' failed                                                                                                                                                                                    
make[2]: *** [third_party/SPIRV-Tools/source/CMakeFiles/SPIRV-Tools-shared.dir/val/validate_cfg.cpp.o] Error 1                                                                              
CMakeFiles/Makefile2:1157: recipe for target 'third_party/SPIRV-Tools/source/CMakeFiles/SPIRV-Tools-shared.dir/all' failed                                                                  
make[1]: *** [third_party/SPIRV-Tools/source/CMakeFiles/SPIRV-Tools-shared.dir/all] Error 2

--- a/source/val/validate_cfg.cpp
+++ b/source/val/validate_cfg.cpp
@@ -1090,8 +1090,9 @@ spv_result_t CfgPass(ValidationState_t& _, const Instruction* inst) {
         return _.diag(SPV_ERROR_INVALID_CFG, inst)
                << "OpReturn can only be called from a function with void "
                << "return type.";
+      _.current_function().RegisterBlockEnd(std::vector<uint32_t>(), opcode);
+      break;
     }
-    // Fallthrough.
     case SpvOpKill:
     case SpvOpReturnValue:
     case SpvOpUnreachable:

